### PR TITLE
Solution for Hoof, Paper, Scissors

### PR DESCRIPTION
**Pull request by ryanchou-dev**
All test cases pass for this solution.

### CODE REVIEW

File: solutions/usaco-688.mdx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+---
+
+## Solution 1
+[Official Analysis](http://www.usaco.org/current/data/sol_hps_bronze_jan17.html)
+
+## Solution 2 - Generating All Permutations
+
+Idea: We can map the games to every possible permuation and find which one gives us the most wins. This works because $1 \leq N \leq 100$
+
+## Implementation
```
Comment (HarryWangATX) [inline solutions/usaco-688.mdx:15]:
```suggestion
### Implementation
```

File: solutions/usaco-688.mdx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+title: Hoof, Paper, Scissors
+author: Ryan Chou
+---
+
+## Solution 1
+[Official Analysis](http://www.usaco.org/current/data/sol_hps_bronze_jan17.html)
+
+## Solution 2 - Generating All Permutations
+
+Idea: We can map the games to every possible permuation and find which one gives us the most wins. This works because $1 \leq N \leq 100$
```
Comment (HarryWangATX) [inline solutions/usaco-688.mdx:13]:
```suggestion
We can map the games to every possible permutation and find the maximum of wins. This works because $1 \leq N \leq 100$
```

Comment (HarryWangATX) [reply solutions/usaco-688.mdx:13]: also add what "permutation"

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
did u add this to problems.json?

File: solutions/usaco-688.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply solutions/usaco-688.mdx:15]: oops, i forgot to add it

File: solutions/usaco-688.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply solutions/usaco-688.mdx:13]: fixing it. 👍

**Comment by ryanchou-dev** [comment]
is the problems.json following this path? usaco-guide/content/extraProblems.json

**Comment by ryanchou-dev** [comment]
oh nvm, that's correct. adding it now. looks like vercel is having trouble because of that.

**Comment by HarryWangATX** [comment]
> oh nvm, that's correct. adding it now. looks like vercel is having trouble because of that.

no, make sure u add it to the json corresponding to ur module.

**Comment by ryanchou-dev** [comment]
> no, make sure u add it to the json corresponding to ur module.

oops. my bad.

File: solutions/usaco-688.mdx [hunk 3]
```diff
@@ -0,0 +1,58 @@
+---
+id: usaco-688
+source: USACO Bronze 2017 January
+title: Hoof, Paper, Scissors
+author: Ryan Chou
+---
+
+## Solution 1
+[Official Analysis](http://www.usaco.org/current/data/sol_hps_bronze_jan17.html)
```
Comment (brad182) [inline solutions/usaco-688.mdx:9]: do [Official Analysis (Java)] (http...)

**Review by brad182 — CHANGES_REQUESTED** [review]

File: solutions/usaco-688.mdx (hunk 3 above)
Comment (maggie-j-liu) [reply solutions/usaco-688.mdx:9]: i think both of those options are fine (?), but i'll check if there's one we want to standardize

**Comment by HarryWangATX** [comment]
OOPS, I didn't realize this problem's not in guide yet. Can u add it to extraProblems.json? Thanks!

**Comment by ryanchou-dev** [comment]
sftlr, fixing it now. 👍

File: solutions/usaco-688.mdx (hunk 3 above)
Comment (brad182) [reply solutions/usaco-688.mdx:9]: i think that the standard should include the language its better that way

**Review by brad182 — APPROVED** [review]

**Comment by ryanchou-dev** [comment]
> Can u add it to extraProblems.json? Thanks!

i added it, lmk if there's anything else. 👍

**Comment by ryanchou-dev** [comment]
is there something wrong with the json, it looks like vercel is having a problem with it.

**Comment by ryanchou-dev** [comment]
oops, i might have added an extra comma.

**Comment by brad182** [comment]
yep thats probably the problem it should build now

**Review by brad182 — APPROVED** [review]

File: solutions/usaco-688.mdx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+## Solution 2 - Generating All Permutations
+
+We can map the games to every possible permutation of the moves (hoof, paper, scissors) and find the maximum of wins. This works because $1 \leq N \leq 100$
+
+### Implementation
```
Comment (dongliu0426) [inline solutions/usaco-688.mdx:15]: add period after "This works because $1 \leq N \leq 100$"

add time complexity below ### Implementation

Comment (ryanchou-dev) [reply solutions/usaco-688.mdx:15]: 👍

**Comment by ryanchou-dev** [comment]
would the code be O(N), since permutations is always of size 6.

File: solutions/usaco-688.mdx (hunk 4 above)
Comment (ryanchou-dev) [reply solutions/usaco-688.mdx]: added it :)

File: solutions/usaco-688.mdx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import sys
+
+sys.stdin = open('hps.in', 'r')
+sys.stdout = open('hps.out', 'w')
+
+n = int(input())
+games = []
+
+for _ in range(n):
+	games.append(list(map(int, input().split())))
```
Comment (SansPapyrus683) [inline solutions/usaco-688.mdx:32]: listcomp maybe?
```py
games = [list(map(int, input().split())) for _ in range(n)]
```

File: solutions/usaco-688.mdx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+# Original Ordering
```
Comment (SansPapyrus683) [inline solutions/usaco-688.mdx:34]:
```suggestion
```
this seems a bit unnecessary

File: solutions/usaco-688.mdx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+	games.append(list(map(int, input().split())))
+
+# Original Ordering 
+# 1 = Hoof, 2 = Paper, 3 = Scissors
+choices = [1, 2, 3]
+
+# in these cases, cow 1 wins
+# hoof beats scissors, 
+# scissors beats paper, 
+# and paper beats hoof
```
Comment (SansPapyrus683) [inline solutions/usaco-688.mdx:41]: don't really think these comments are necessary? actually maybe it'd be better to just replace them with 
```py
# hoof -> scissors -> paper -> hoof
```

File: solutions/usaco-688.mdx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+pers = list(permutations(choices, 3))
+
+maxwins = 0
+
+for j in pers:
+	wins = 0
+	for i in games:
+		selection = [j[i[0]-1], j[i[1]-1]]
+		if selection in scenarios:
+			wins+=1
```
Comment (SansPapyrus683) [inline solutions/usaco-688.mdx:54]: add some whitespace between operators plz?

File: solutions/usaco-688.mdx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+# hoof beats scissors, 
+# scissors beats paper, 
+# and paper beats hoof
+scenarios = [[1, 3], [3, 2], [2, 1]]
+
+# generates all permutations
+pers = list(permutations(choices, 3))
+
+maxwins = 0
+
```
Comment (SansPapyrus683) [inline solutions/usaco-688.mdx:48]: 46 and 48 seem like unnecessary whitespace

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
some explanation in the for loop would be nice? but maybe i'm just dumb

File: solutions/usaco-688.mdx (hunk 5 above)
Comment (ryanchou-dev) [reply solutions/usaco-688.mdx:32]: sure!

Comment (ryanchou-dev) [reply solutions/usaco-688.mdx:32]: adding it now.

Comment (ryanchou-dev) [reply solutions/usaco-688.mdx]: 👍

File: solutions/usaco-688.mdx (hunk 7 above)
Comment (ryanchou-dev) [reply solutions/usaco-688.mdx:39]: thanks!

File: solutions/usaco-688.mdx (hunk 9 above)
Comment (ryanchou-dev) [reply solutions/usaco-688.mdx]: > some explanation in the for loop would be nice? but maybe i'm just dumb

thanks for the suggestion, adding it now. :)

File: solutions/usaco-688.mdx [hunk 10]
```diff
@@ -49,7 +47,7 @@ for j in pers:
 	for i in games:
 		selection = [j[i[0]-1], j[i[1]-1]]
```
Comment (SansPapyrus683) [inline solutions/usaco-688.mdx:46]: uh whitespace between these ops?

Comment (ryanchou-dev) [reply solutions/usaco-688.mdx:46]: oops, my bad

**Comment by ryanchou-dev** [comment]
I've added the explanation and whitespace. 👍

**Review by SansPapyrus683 — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]

File: solutions/usaco-688.mdx [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+## Solution 1
+[Official Analysis](http://www.usaco.org/current/data/sol_hps_bronze_jan17.html)
+
+## Solution 2 - Generating All Permutations
+
+We can map the games to every possible permutation of the moves (hoof, paper, scissors) and find the sequence which gives us the maximum wins. This works because $1 \leq N \leq 100$.
+
+### Implementation
+
+**Time Complexity: ** $\mathcal{O}(n)$.
```
Comment (HarryWangATX) [inline solutions/usaco-688.mdx:17]:
```suggestion
**Time Complexity:** $\mathcal{O}(N)$

```

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

**Review by maggie-j-liu — APPROVED** [review]

**Review by HarryWangATX — APPROVED** [review]
